modin/pandas/dataframe.py

```diff
@@ -1090,7 +1090,20 @@ def cumsum(self, axis=None, skipna=True, *args, **kwargs):
             The cumulative sum of the DataFrame.
         """
         axis = pandas.DataFrame()._get_axis_number(axis) if axis is not None else 0
-        self._validate_dtypes(numeric_only=True)
+        dtype = self.dtypes[0]
+        # Checks to make sure that all dtypes are of the same category as the first
+        if axis and not all(
```

Comment:
I think this ends up checking the type of `self.dtypes[0]` for every iteration, but you really only need to get the dtype of the first item once and then check if the rest of the columns have that same dtype.